app/services/notifications/dead_link.rb

```diff
@@ -0,0 +1,54 @@
+module Notifications
+  class DeadLink
+    include Rails.application.routes.url_helpers
+
+    def initialize(flag)
+      @flag = flag
+    end
+
+    def self.call(flag)
+      new(flag).call
+    end
+
+    def call
+      notification.deliver_later(project_submission_owner)
+    end
+
+    private
+
+    def notification
+      FlagNotification.with(flag: flag, message: message, url: url)
+    end
+
+    def project_submission_owner
+      flag.project_submission.user
+    end
+
+    # rubocop:disable Layout/LineLength
+    def message
+      "Hey #{username}, heads up your project submission for #{lesson_name} was flagged because it has a broken link. " \
+        'We aim to keep the submission lists useful and so we remove any submissions with broken links often. ' \
+        'When you get a moment, can you update your submission with the correct links for us please? ' \
+        "Otherwise we will have to remove it on the #{date}."
+    end
+    # rubocop:enable Layout/LineLength
+
+    def url
+      lesson_url(flag.project_submission.lesson, only_path: true)
+    end
+
+    def username
+      flag.project_submission.user.username
+    end
+
+    def lesson_name
+      flag.project_submission.lesson.title
+    end
+
+    def date
+      7.days.from_now.strftime('%d %b %Y')
+    end
+
+    attr_reader :flag
```

Comment:
Can we put this `attr_reader` just under the private keyword please? just to keep it consistent with the other services.